[CHANGELOG] Prepare v5.0.0-alpha.16 (#23515)
mui-org/material-ui: CHANGELOG.md

### [Versions](https://material-ui.com/versions/)

## 5.0.0-alpha.16

###### _Nov 14 2020_

Big thanks to the 34 contributors who made this release possible. Here are some highlights ✨:

> - 📅 Migrate to the date picker in the lab (#22692) @dmtrKovalenko.

Reviewer: ```suggestion
- 📅 Migrate the date picker to the lab (#22692) @dmtrKovalenko.
```

>   We have integrated the components inside the code infrastructure. In the coming days, we will migrate all the issues from [material-ui-pickers](https://github.com/mui-org/material-ui-pickers) and archive the repository. This migration will help provide first-class support for the date picker components. The component will stay in the lab as long as necessary to reach the high expectations bar we have for core components. You can find the [new documentation here](https://next.material-ui.com/components/pickers/).

Reviewer: ```suggestion
  We have integrated the components with the code infrastructure. Next we will migrate all the GitHub issues from [material-ui-pickers](https://github.com/mui-org/material-ui-pickers) and archive the repository. This migration will help provide first-class support for the date picker components. The component will stay in the lab as long as necessary to reach the high quality bar we have for core components. You can find the [new documentation here](https://next.material-ui.com/components/pickers/).
```

>   While the sources are currently hosted in the [main repository](https://github.com/mui-org/material-ui), we might move them to the [x repository](https://github.com/mui-org/material-ui-x), depending on what is easier for the commercial date range picker. The date picker will stay open source no matter what.

Reviewer: ```suggestion
  While the source code is currently hosted in the [main repository](https://github.com/mui-org/material-ui), we might move it to the [x repository](https://github.com/mui-org/material-ui-x) in the future, depending on what is easier for the commercial date range picker. The date picker will stay open source no matter what.
```

> - 📚 Revamp the documentation for [the system](https://next.material-ui.com/system/basics/). The System contains CSS utilities for rapidly implementing custom designs. It now emphasizes the `sx` prop. It's ideal for adding one-off styles, e.g. padding. When pushed to its limits, it can be used to implement quickly a complete marketing page.

Reviewer: ```suggestion
- 📚 Revamp the documentation for [the system](https://next.material-ui.com/system/basics/). The System contains CSS utilities. The documentation now promotes the use of the `sx` prop. It's ideal for adding one-off styles, e.g. padding, but when pushed to its limits, it can be used to implement quickly a complete page.
```

- 👩‍🎨 Upgrade emotion to v11 (#23007) @mnajdova.
- And many more 🐛 bug fixes and 📚 improvements.

### `@material-ui/core@v5.0.0-alpha.16`

#### Breaking changes

- [TextField] Change default variant from standard to outlined (#23503) @mbrookes
  Standard has been removed from the Material Design guidelines. [This codemod](https://github.com/mui-org/material-ui/tree/next/packages/material-ui-codemod#textfield-variant-prop) will automatically update your code.

  ```diff
  -<TextField value="Standard" />
  -<TextField value="Outlined" variant="outlined" />
  +<TextField value="Standard" variant="standard" />
  +<TextField value="Outlined" />
  ```

- [Autocomplete] Remove `debug` in favor of `open` and dev tools (#23377) @eps1lon
  There are a couple of simpler alternatives: `open={true}`, Chrome devtools ["Emulate focused"](https://twitter.com/sulco/status/1305841873945272321), or React devtools props.

#### Changes

> - [Autocomplete] Use Popper with when disablePortal (#23263) @eps1lon

Reviewer 2: ```suggestion
- [Autocomplete] Use Popper when `disablePortal` (#23263) @eps1lon
```
I sometimes change what I want to say mid-sentence which leads to these confusing statements sometimes :smile:

- [Tooltip] Long press select text on iOS (#23466) @hmaddisb
- [Tooltip] Unexpected behaviour onOpen/onClose (#23482) @brorlarsnicklas

### `@material-ui/lab@v5.0.0-alpha.16`

- [DatePicker] Migrate to the lab #22692 @dmtrKovalenko

### `@material-ui/system@v5.0.0-alpha.16`

- [system] Add typography prop that will pull from theme.typography (#23451) @mnajdova
- [system] Create separate margin and padding functions (#23452) @mnajdova
- [system] Export styleFunctionSx and improve signature (#23397) @mnajdova
- [system] Merge breakpoints in correct order (#23380) @mnajdova
- [system] Remove css utility in favor of sx (#23454) @mnajdova
- [system] Warn for spacing when non integer value is used with theme.spacing array (#23460) @mnajdova

### `@material-ui/styled-engine@v5.0.0-alpha.16`

- [styled-engine] Upgrade emotion to 11 RC (#23007) @mnajdova

### `@material-ui/unstyled@v5.0.0-alpha.16`

- [Slider] Extract slots as standalone components (#22893) @mnajdova

### `@material-ui/codemod@v5.0.0-alpha.16`

- [TextField] Change default variant from standard to outlined (#23503) @mbrookes

### Docs

- [docs] Allow to host code in a different repo (#23390) @oliviertassinari

> - [docs] Avoid confusing nav items with disabled items (#23283) @eps1lon

Reviewer 2: This can be removed since it was reverted out-of-band.

> - [docs] Avoid confusing nav items with disabled items (#23283) @eps1lon

Reviewer: ```suggestion
```
